### PR TITLE
invitation link #1116

### LINKED ISSUE #1116: Issues on account creation flow (2/2)
- [x] when a user is created by invitation (add a mail that does not exists), they can signin without firstname and lastname and the page to fill details won't show up. The page should appear until they filled the required information.
  - [x] if you refresh once you'll get the page, but a 'no uid' error when you fill the form
  - [x] if you refresh twice you might be able to fill the form and get in
- [ ] when a user invites another user to their organizatin, the "pending members" page doesn't refresh every time

Relates to #738

### PR DESCRIPTION
**Pull request by Maxstgt**
close #1116 
user gets invitation link with inivtationId for mapping
- [ ] tested with sendgridAPI key

### CODE REVIEW

File: apps/backend-functions/src/assets/mail-templates.ts [hunk 1]
```diff
@@ -2,23 +2,19 @@
...
-  `
-  You've been invited to a project on the Blockframes Platform!
-
-  login: ${email}\n
-  password: ${password}\n
-
-  Click on the following link to login to your account and join the project:
-  ${appUrl}${USER_WELCOME_PATH}
-  `;
+// ------------------------- //
```
Comment (lsenta) [inline apps/backend-functions/src/assets/mail-templates.ts]: Please don't ascii art in the code, either document the functions or move to separate files.

File: apps/backend-functions/src/internals/firebase.ts [hunk 2]
```diff
@@ -23,3 +24,11 @@ export async function getUserMail(userId: string): Promise<string | undefined> {
   const user = await admin.auth().getUser(userId);
   return user.email;
 }
+
+/**
+ * Gets the user email for the user corresponding to a given `uid`.
+ * Throws if the user does not exists.
+ */
+export function getUser(userId: string): Promise<PublicUser> {
+  return db.doc(`/users/${userId}`).get().then(user => user.data() as PublicUser);
```
Comment (lsenta) [inline apps/backend-functions/src/internals/firebase.ts]: Stay consistent with the codebase, don't use promise's `then`'ing, use `async`

File: apps/backend-functions/src/main.ts [hunk 3]
```diff
@@ -61,6 +61,14 @@ export const onUserCreate = functions.auth
 export const findUserByMail = functions.https
   .onCall(users.findUserByMail);
 
+/** Trigger: REST call to send a verify email to a user. */
+export const sendVerifyEmail = functions.https
```
[the remark this answers is not in the document]
Comment (pldespaigne) [reply apps/backend-functions/src/main.ts]: @Jackseed

File: env/env.ci.ts [hunk 4]
```diff
@@ -64,3 +64,18 @@ export const relayer = {
 
 export const backupBucket = 'ci-backups';
 export const sendgridAPIKey = '';
+
+// Sendgrid Emails
+
+export const templateIds = {
```
Comment (lsenta) [inline env/env.ci.ts]: 👍

File: libs/auth/src/lib/+state/auth.service.ts [hunk 5]
```diff
@@ -151,9 +153,10 @@ export class AuthService {
    * @email find the user with this email. If email doesn't match with an existing user,
    * create a user with this email address.
    */
-  public async getOrCreateUserByMail(email: string): Promise<User> {
+  // public async getOrCreateUserByMail(email: string): Promise<User> {
+  public async getOrCreateUserByMail(email: string, invitationId?: string): Promise<User> {
     const f = firebase.functions().httpsCallable('getOrCreateUserByMail');
-    return f({ email }).then(matchingEmail => matchingEmail.data);
+    return f({ email, invitationId }).then(matchingEmail => matchingEmail.data);
```
[the remark this answers is not in the document]
Comment (lsenta) [reply libs/auth/src/lib/+state/auth.service.ts]: You don't need the invitationId here
Where is it used in the backend?

File: apps/backend-functions/src/main.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (pldespaigne) [reply apps/backend-functions/src/main.ts]: We are not leaving firebase auth framework, but we are definitely going to pay those emails

File: libs/organization/src/lib/pages/organization-loading/organization-loading.component.ts [hunk 6]
```diff
@@ -10,13 +11,14 @@ import { AuthQuery } from "@blockframes/auth";
...
   constructor(
+    private router: Router,
     private query: AuthQuery,
   ) {}
 
   ngOnInit() {
     this.query.user$.subscribe(user => {
       if (!!user.orgId) {
-        window.location.assign('/');
+        this.router.navigateByUrl(`/`);
```
Comment (ioaNikas) [inline libs/organization/src/lib/pages/organization-loading/organization-loading.component.ts]: Use regular quotes if you doesn't need backquotes.

File: apps/backend-functions/src/users.ts [hunk 7]
```diff
@@ -122,10 +166,18 @@ const getOrCreateUserByMail = async (
...
-export { onUserCreate, findUserByMail, findOrgByName, getOrCreateUserByMail };
+export {
+  onUserCreate,
+  findUserByMail,
+  findOrgByName,
+  getOrCreateUserByMail,
+  startVerifyEmailFlow,
+  startResetPasswordEmailFlow,
+  startWishlistEmailsFlow,
+};
```
[the remark this answers is not in the document]
Comment (pldespaigne) [reply apps/backend-functions/src/users.ts]: not sure why it's like that, @lsenta did that 6 month ago, so I just add my new functions there.
But it's the only place in the backend functions where we do that, I will change to the classical export if there is no particular reason for this export form

**Pull request merged**